### ISSUE TITLE
Unreachable `match` wildcard case is not reported

### ISSUE THREAD

**Issue by injust** [python/mypy#20251]
**Bug Report**

When a `match` statement on an enum is already exhaustive, mypy does not consider the wildcard case to be unreachable.

**To Reproduce**

https://mypy-play.net/?mypy=latest&python=3.14&enable-error-code=exhaustive-match&flags=warn-unreachable&gist=0aab275a6ddc8e9d83ea8bc676debf07

```python
from enum import Enum, auto


class Thing(Enum):
    ONE = auto()


def main(foo: Thing) -> None:
    match foo:
        case Thing.ONE:
            pass
        case _:
            pass
```

**Expected Behavior**

The wildcard case is unreachable and should be reported as an error.

**Actual Behavior**

```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 1.18.2
- Mypy command-line flags: `--warn-unreachable --enable-error-code=exhaustive-match`
- Python version used: 3.14

**Comment by A5rocks** [comment]
Possible duplicate of https://github.com/python/mypy/issues/20142 though single-value enum logic is sometimes different (sadly).

**Comment by A5rocks** [comment]
Hmm maybe if you replace `pass` with eg `print("hi")` it might warn about unreachability? Mypy doesn't flag `pass` for unreachability...

(I cannot check right now)

**Comment by injust** [comment]
> Hmm maybe if you replace `pass` with eg `print("hi")` it might warn about unreachability? Mypy doesn't flag `pass` for unreachability...

Yup, that triggered the unreachability error.

I guess this is working as intended then?

**Comment by A5rocks** [comment]
Yeah
